SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt

```diff
@@ -0,0 +1,135 @@
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val workflowResult = getResult {
+                            workflowService.createWorkflow(
+                                postWorkflowBankModel = PostWorkflowBankModel(
+                                    type = PostWorkflowBankModel.Type.plaid,
+                                    kind = PostWorkflowBankModel.Kind.create,
+                                    customerGuid = customerGuid,
+                                    externalBankAccountGuid = "3aa2bc487ae469bbc03cb42cf72e3aa3",
+                                    language = PostWorkflowBankModel.Language.en,
+                                    linkCustomizationName = "default",
+                                    androidPackageName = "app.cybrid.sdkandroid"
+                                )
+                            )
+                        }
+                        Logger.log(LoggerEvents.DATA_FETCHED, "Create - Workflow")
+                        workflowResult.let {
+                            if (isSuccessful(it.code ?: 500)) {
+                                workflowJob = Polling { fetchWorkflow(guid = workflowResult.data?.guid!!) }
+                            }
+                        }
+                    }
+                    waitFor.await()
+                }
+            }
+        }
+    }
+
+    fun fetchWorkflow(guid: String) {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    scope.launch {
+                        val workflowResult = getResult {
+                            workflowService.getWorkflow(workflowGuid = guid)
+                        }
+                        Logger.log(LoggerEvents.DATA_REFRESHED, "Fetch - Workflow")
+                        workflowResult.let {
+                            if (isSuccessful(it.code ?: 500)) {
+                                checkWorkflowStatus(workflow = workflowResult.data!!)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    suspend fun createExternalBankAccount(publicToken: String, account: LinkAccount?) {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val externalBankAccountResult = getResult {
+                            externalBankAccountsService.createExternalBankAccount(
+                                postExternalBankAccountBankModel = PostExternalBankAccountBankModel(
+                                    name = account?.name ?: "",
+                                    accountKind = PostExternalBankAccountBankModel.AccountKind.plaid,
+                                    asset = "USD",
```

Comment:
This needs to be set somehow by the caller. It will depend on the setting of the bank.

Reply:
Done!